lib/index.js

```diff
@@ -717,7 +719,13 @@ exports.merge = (schemas, options = {}) => {
 		if (mergedSchemas.anyOf && mergedSchemas.anyOf.length === 1) {
 			const [ anyOf ] = mergedSchemas.anyOf
 
-			return merge(omit(mergedSchemas, 'anyOf'), anyOf)
+			// Deduplicate required array values
+			const merged = merge(omit(mergedSchemas, 'anyOf'), anyOf)
+			return traverse(merged).forEach(function (_node) {
+				if (this.key === 'required' && isArray(_node) && _node.length > 1) {
+					this.update(uniq(_node))
+				}
+			})
```

Comment:
I do have a performance concern when it comes to traversing the entire merged schema object, but we need to account for complex schema objects. Also unsure if deduplicating should be done for all arrays and not only `required` arrays.